frontend/src/tests/instructor-permission-test.js

```diff
@@ -15,14 +15,167 @@ export function instructorsPermissionTests(api) {
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiGET(`/debug/users`);
+        expect(resp).toHaveStatus("success");
+
+        resp.payload.forEach((user) => {
+            if (user.utorid === instructorOnlyUserData.utorid) {
+                expect(user.roles).toEqual(["instructor"]);
+                instructorOnlyUser = user;
+            }
+        });
+
+        expect(instructorOnlyUser).toBeDefined();
     }, 30000);
 
+    it("assigning to be instructor of a course grants instructor role", async () => {
+        await restoreDefaultUser();
+        const emptyRoleUserUtorid =
+            "course_assign_grant_instructor_test_user_utorid";
+        const emptyRoleUserData = {
+            utorid: emptyRoleUserUtorid,
+            roles: [],
+        };
+
+        let resp = await apiPOST(`/debug/users`, emptyRoleUserData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiGET(`/debug/users`);
+        expect(resp).toHaveStatus("success");
+
+        let emptyRoleUserExists = false;
+        resp.payload.forEach((user) => {
+            if (user.utorid === emptyRoleUserData.utorid) {
+                emptyRoleUserExists = true;
+            }
+        });
+
+        expect(emptyRoleUserExists).toBeTruthy();
+
+        const instructorData = {
+            utorid: emptyRoleUserUtorid,
+            first_name: "course_assign_grant_instructor_test_first_name",
+            last_name: "course_assign_grant_instructor_test_last_name",
+            email: "course_assign_grant_instructor_test_email",
+        };
+        resp = await apiPOST(`/admin/instructors`, instructorData);
+        expect(resp).toHaveStatus("success");
+
+        resp = await apiGET(`/admin/instructors`);
+        expect(resp).toHaveStatus("success");
+        let instructorId;
+        resp.payload.forEach((instructor) => {
+            if (instructor.utorid === emptyRoleUserUtorid) {
+                instructorId = instructor.id;
+            }
+        });
+        expect(instructorId).toBeDefined();
+
+        const defaultTestPosition = {
+            position_code: "course_assign_grant_instructor_test_position_code",
+            position_title:
+                "course assign grant instructor test position title",
+            hours_per_assignment: 1.0,
+            contract_template_id: existingContractTemplateId,
+            duties: "Tutorials",
+            instructor_ids: [instructorId],
+        };
+
+        // upon successfully making the following position upsert, the no-role user should be granted instructor role
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/positions`,
+            defaultTestPosition
+        );
+        expect(resp).toHaveStatus("success");
+
+        // set active user to trigger role update
+        resp = await apiPOST(`/debug/active_user`, emptyRoleUserData);
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET(`/debug/active_user`);
```

Comment:
There's no reason for all this to be taking place for the active user. In fact, that's kind of weird...the active user should be an `admin` the whole time, and a new user should be created and retrieved.

Reply:
Ditto